certification/internal/policy/operator/deployable_by_olm.go

```diff
@@ -178,30 +182,37 @@ func (p *DeployableByOlmCheck) installedCSV(operatorData OperatorData, openshift
 	defer cancel()
 
 	installedCSVDone := make(chan string, 1)
+	contextTimeOut := make(chan error, 1)
 
+	var installedCSV string
 	// query API server for the installed CSV field of the created subscription
-	go func() {
+	go func(ctx context.Context) string {
 		defer close(installedCSVDone)
 		for {
-			time.Sleep(2 * time.Second)
-			log.Debug("Waiting for Subscription.status.installedCSV to become ready...")
-			subs, _ := openshiftengine.GetSubscription(operatorData.App, cli.OpenshiftOptions{Namespace: operatorData.InstallNamespace})
-			installedCSV := subs.Status.InstalledCSV
-			// if the installedCSV field is present, stop the querying
-			if len(installedCSV) > 0 {
-				log.Debug(fmt.Sprintf("Subscription.status.installedCSV is %s", installedCSV))
-				installedCSVDone <- installedCSV
+			select {
+			case <-ctx.Done():
+				log.Error("failed to fetch Subscription.status.installedCSV: ", ctx.Err())
+				contextTimeOut <- ctx.Err()
+			default:
+				time.Sleep(2 * time.Second)
+				log.Debug("Waiting for Subscription.status.installedCSV to become ready...")
+				subs, _ := openshiftengine.GetSubscription(operatorData.App, cli.OpenshiftOptions{Namespace: operatorData.InstallNamespace})
```

Comment:
I would leave this outside of the select statement. The select should either exit or sleep. Everything else is a part of the "main" routine

Reply:
I take it back. Again, I was reading this wrong.

Reply:
Here's what I understand now. With the context you are passed in, that's where you should add the WithTimeout... so, ctx, cancel := context.WithTimeout(ctx, 180 * time.Second), then defer cancel(). I'll add another comment elsewhere.